sdk/include/opentelemetry/sdk/resource/semantic_conventions.h

```diff
@@ -0,0 +1,139 @@
+#include <stdint.h>
+#include <iostream>
+#include <type_traits>
+#include <unordered_map>
+#include "opentelemetry/version.h"
+
+// typedef uint32_t Tag;
+
+static constexpr uint32_t hashCode(const char *str, uint32_t h = 0)
+{
+  return (uint32_t)(!str[h] ? 5381 : ((uint32_t)hashCode(str, h + 1) * (uint32_t)33) ^ str[h]);
```

Comment:
Is there any reference for this hash function? Put a link here if so.

Reply:
Good point. Have moved this function to `api/include/opentelemetry/common/string_util.h` and the reference of the hash algorithm.